Add `MaskedInput` docs (#1476)
hashicorp/design-system: website/docs/components/form/masked-input/partials/guidelines/overview.md

> A masked input field is a special input that visually obfuscates characters to protect sensitive information by masking them with a circular shape.

Reviewer: Should "Masked Input" be capitalized?

Reviewer 2: Probably, yes.

!!! Warning

**Important to know**

> This component is meant for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained through other means (eg. copying it from the input, via JavaScript, via the developer tools, etc.).

Reviewer 3: ```suggestion
This component is meant for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained through other means (e.g., copying it from the input, via JavaScript, via the developer tools, etc.).
```

> This component it's meant to be used for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained with other means (eg. copying it from the input, via JavaScript, via the developer tools, etc.).

Reviewer: "through/by other means" would probably be better here

```suggestion
This component it's meant to be used for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained through other means (eg. copying it from the input, via JavaScript, via the developer tools, etc.).
```

Reviewer 4: There are other errors so I added the below suggested edits

> This component it's meant to be used for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained with other means (eg. copying it from the input, via JavaScript, via the developer tools, etc.).

Reviewer 4: ```suggestion
This component is meant for **visual obfuscation** only. Consumers should be aware that the hidden text value could still be obtained through other means (eg. copying it from the input, via JavaScript, via the developer tools, etc.).
```
